packages/storage-plus/src/indexed_map.rs

```diff
@@ -175,7 +175,8 @@ mod test {
     }
 
     struct DataIndexes<'a> {
-        pub name: MultiIndex<'a, Data>,
+        // Second arg is for storing pk
+        pub name: MultiIndex<'a, (PkOwned, PkOwned), Data>,
```

Comment:
It would be a good test to see if it is possible (or not) to use &str or &[u8] in secondary indexes.
Even a failure and then a commented line saying why it is not possible is a clear guide.

Reply:
I think it should be possible. Maybe not `&[u8]` but `[u8]` (i.e. an owned slice). Related to the other comments. If you want me to address this here, I can give it a try during the week, or better, next week.

Reply:
If you could just try it and see compiler errors that would be nice.
If it is non-trivial, just comment it out and make an issue for later

Reply:
Here are the errors, when changing the `name` `MultiIndex` to use `&[u8]` in the `indexed_map` tests:
```
   Compiling cw-storage-plus v0.5.0 (/usr2/home/mauro/work/cosmwasm-plus/packages/storage-plus)
error[E0495]: cannot infer an appropriate lifetime for autoref due to conflicting requirements
   --> packages/storage-plus/src/indexed_map.rs:209:50
    |
209 |             name: MultiIndex::new(|d, k| (d.name.as_ref(), &*k), "data", "data__name"),
    |                                                  ^^^^^^
    |
note: first, the lifetime cannot outlive the anonymous lifetime #1 defined on the body at 209:35...
   --> packages/storage-plus/src/indexed_map.rs:209:35
    |
209 |             name: MultiIndex::new(|d, k| (d.name.as_ref(), &*k), "data", "data__name"),
    |                                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
note: ...so that reference does not outlive borrowed content
   --> packages/storage-plus/src/indexed_map.rs:209:43
    |
209 |             name: MultiIndex::new(|d, k| (d.name.as_ref(), &*k), "data", "data__name"),
    |                                           ^^^^^^
note: but, the lifetime must be valid for the lifetime `'a` as defined on the function body at 207:18...
   --> packages/storage-plus/src/indexed_map.rs:207:18
    |
207 |     fn build_map<'a>() -> IndexedMap<'a, &'a [u8], Data, DataIndexes<'a>> {
    |                  ^^
note: ...so that the expression is assignable
   --> packages/storage-plus/src/indexed_map.rs:216:9
    |
216 |         IndexedMap::new("data", indexes)
    |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    = note: expected `indexed_map::IndexedMap<'a, &'a [u8], _, DataIndexes<'a>>`
               found `indexed_map::IndexedMap<'_, &[u8], _, DataIndexes<'_>>`

error: aborting due to previous error
```